Automattic/jetpack
# Jetpack Social: Support VideoPress videos in media picker

## Proposed changes:

Currently, it is not possible to pick a VideoPress video from the Jetpack Social media picker. They don't show up in the media picker and uploading a new video gives an unsupported file type error, because VideoPress videos have the `video/videopress` mime type.

This PR changes two things:

1. It adds support for the `video/videopress` mime type. This allows them to show up in the media picker.
2. When a user picks/uploads a VideoPress video, we parse the metadata slightly differently. The length of the video gets converted from the VideoPress meta data, and the original mime type is (naively) taken from the source URL's file extension.

### Other information:


## Jetpack product discussion


1203420104361773-as-1203821165244931/f

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






* Create a new JN site with bleeding edge Jetpack with the Beta plugin.
* Enable sharing and add a Twitter or Facebook connection.
* From My Jetpack, enable VideoPress and purchase a plan.
* Add enhanced publishing support to your JN site.
* Create a new post and try to attach a video in the Jetpack Social sidebar.
* It should give an error about the file type.
* Now, switch to this branch with the Jetpack Beta plugin, and try adding a video again.
* The video uploaded earlier should now be visible from the media picker, and uploading a new one should work as well.
* Both ways should show a video preview when picked.
* Make sure the post saves correctly.
